### PR TITLE
[feature] adding-cliff-framework

### LINKED ISSUE #18: feat: framework - using a cmd library/framework to handle app execution
As developers, i want use libraries like [cliff](https://docs.openstack.org/developer/cliff/), [click](http://click.pocoo.org/5/), ect. This will allow for a standardization of how to create future command and will help improve flag handling.

### PR DESCRIPTION
**Pull request by gardlt**
* fixing some logging issues
* added license
* crated more more example deploys: wordpress, keystone deploy
* added editconfig for consistent standards

**What is the purpose of this pull request?**:

**What issue does this pull request address?**: Fixes #18 

**Notes for reviewers to consider**:

**Specific reviewers for pull request**:

### CODE REVIEW

**Comment by gardlt** [comment]
this one is dependent on #26 this will need to be rebased

File: armada/log.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        return self.LEVEL_COLORS[record.levelno] + res + '\033[m'
+
+
+def set_console_formatter(**formatter_kwargs):
+    formatter_kwargs.setdefault(
+        'fmt', '%(asctime)s %(name)-12s %(levelname)-8s %(message)s')
+    formatter_kwargs.setdefault('datefmt', '%m-%d %H:%M')
+    root_logger = logging.getLogger('')
+    for handler in root_logger.handlers:
+        if handler.__class__ is logging.StreamHandler:  # Skip subclasses
```
Comment (g0r1v3r4) [inline armada/log.py]: should this be something like `instanceOf()`? or does the `is` keyword pickup on instance equality?

File: armada/log.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            console_handler = handler
+            # Skip if not a tty (default ssh, redirect, ...)
+            isatty = getattr(handler.stream, 'isatty', None)
+            if isatty is None or not isatty():
+                continue
+            break
+    else:
```
Comment (g0r1v3r4) [inline armada/log.py]: the `else` indentation looks like aligns with the `for` and not the `if`

File: README.md [hunk 3]
```diff
@@ -95,7 +95,7 @@ endpoints: &endpoints
 And re-run armada, we will notice it will upgrade the keystone release, instead of install it on this pass, as well as report back the values changes as a unified diff.  A unified diff for any template changes would also be shown had those occurred.
 
 ```
-alan@hpdesktop:~/Workbench/att/attcomdev/armada$ armada -c examples/armada.yaml
+alan@hpdesktop:~/Workbench/att/attcomdev/armada$ armada apply examples/armada.yaml
```
Comment (gardlt) [inline README.md]: yeah will remove to just $

File: README.md [hunk 4]
```diff
@@ -62,13 +62,13 @@ Before using armada we need to check a few things:
 To run armada, simply supply it with your YAML based intention for any number of charts:
 
 ```
-aramda -c examples/armada.yaml
+aramda apply examples/armada.yaml [--debug |
```
Comment (gardlt) [inline README.md]: done

File: armada/log.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        res = Formatter.format(self, record)  # old-style class on 2.6
+        return self.LEVEL_COLORS[record.levelno] + res + '\033[m'
+
+
+def set_console_formatter(**formatter_kwargs):
+    formatter_kwargs.setdefault(
+        'fmt', '%(asctime)s %(name)-12s %(levelname)-8s %(message)s')
+    formatter_kwargs.setdefault('datefmt', '%m-%d %H:%M')
+    root_logger = logging.getLogger('')
+    for handler in root_logger.handlers:
```
Comment (gardlt) [inline armada/log.py]: i dont belive that this would work better in an while. can you explain why?

File: armada/log.py (hunk 1 above)
Comment (gardlt) [inline armada/log.py]: is covers it

File: armada/log.py (hunk 2 above)
Comment (gardlt) [inline armada/log.py]: yes its a for else statement

**Comment by alanmeadows** [comment]
Catching up on the pull requests here -- seems this one has languished after feedback?

**Pull request closed without merging**